src/ActivityHeartbeatingCancellation/Program.cs

```diff
@@ -1,17 +1,21 @@
 using Microsoft.Extensions.Logging;
 using Temporalio.Client;
+using Temporalio.Client.EnvConfig;
 using Temporalio.Exceptions;
 using Temporalio.Worker;
 using TemporalioSamples.ActivityHeartbeatingCancellation;
 
 // Create a client to localhost on default namespace
-var client = await TemporalClient.ConnectAsync(new("localhost:7233")
+var connectOptions = ClientEnvConfig.LoadClientConnectOptions();
+if (string.IsNullOrEmpty(connectOptions.TargetHost))
```

Comment:
Would recommend `connectOptions.TargetHost ??= "localhost:7233";` which I believe works. Also, I assume `connectOptions.TargetHost` is not set to an empty string if it is unset in env config, correct?

Reply:
> The class does accept null values, just not in that constructor, so you should use the other constructor, e.g. var options = new TemporalClientConnectOptions() { TargetHost = Address };

Ah I missed this.

> But I'd be ok if we do this for now and come back and ??= it when we represent Address as null properly. Would we prefer that approach? (if so, we can create an issue and I'll mark approved no this PR)

I'd prefer this. I can open a new PR on sdk and samples, would prefer to not be blocked on another release.